### PR TITLE
Expose public methods to internal apis.

### PR DESCRIPTION
**Pull request by Drapersniper**
### Type

- [ ] Bugfix
- [x] Enhancement
- [x] New feature

##Note
If you currently feel like you would benefit from a public method let me know and ill look into adding it to this PR.

### Description of the changes
- Add some methods to interact with the blocklist/allowlist.
- Restore the old `bot.counter` property with several safeguards

### CODE REVIEW

**Comment by teddyrasmussen** [comment]
A public method I would like to see is a way to get the song playing in a guild from the Audio cog.

**Comment by Drapersniper** [comment]
> A public method I would like to see is a way to get the song playing in a guild from the Audio cog.

Not feasible, until the api is done. (You do know you can use Red-LavaLink to get the player and from there get the track from the player right?)

**Comment by teddyrasmussen** [comment]
> Not feasible, until the api is done. (You do know you can use Red-LavaLink to get the player and from there get the track from the player right?)
How can I do that?

**Comment by aikaterna** [comment]
> How can I do that?
Answered in Red #coding

File: redbot/core/utils/_internal_utils.py [hunk 1]
```diff
@@ -316,3 +320,138 @@ async def fetch_latest_red_version_info() -> Tuple[Optional[VersionInfo], Option
...
+    def get(self, cog: Cog, counter: str) -> int:
+        return self.get_raw(cog.qualified_name, counter)
+
+    @final
+    def get_raw(self, cog_qualified_name: str, counter: str) -> int:
+        return self.__getitem__((cog_qualified_name, counter,))
+
+    @final
+    def inc(self, cog: Cog, counter: str) -> int:
+        return self.tick_raw(cog.qualified_name, counter)
```
Comment (TrustyJAID) [inline redbot/core/utils/_internal_utils.py]: Should this be `inc_raw` instead?

File: redbot/core/events.py [hunk 2]
```diff
@@ -375,13 +377,59 @@ async def on_guild_remove(guild: discord.Guild):
...
+        bot.counter._inc_core_raw("Red_Core", "on_raw_reaction_add")
+
+    @bot.event
+    async def on_trivia_end(session):
+        bot.counter._inc_core_raw("Red_Core", "on_trivia_end")
+
+    @bot.event
+    async def on_filter_message_delete(message, hits):
+        bot.counter._inc_core_raw("Red_Core", "on_filter_message_delete")
+
```
Comment (Drapersniper) [inline redbot/core/events.py:477]: This Part of the PR is not final ... i just added here for the purpose of tracking the events monitored/dispatched by core and core cogs.... adding it here would make it easier to not have duplicated counts... but may not be ideal.

**Comment by Zephyrkul** [comment]
Requesting a public version of core.utils.mod's `is_allowed_by_hierarchy` (with a better signature and without depending on Mod's config, as per #coding)

**Pull request closed without merging**